do/droplets.go

```diff
@@ -144,6 +144,25 @@ func (i *instances) CurrentNodeName(hostname string) (types.NodeName, error) {
 // InstanceExistsByProviderID returns true if the instance for the given provider id still is running.
 // If false is returned with no error, the instance will be immediately deleted by the cloud controller manager.
 func (i *instances) InstanceExistsByProviderID(providerID string) (bool, error) {
+	id, err := dropletIDFromProviderID(providerID)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = dropletByID(context.TODO(), i.client, id)
+	if err == nil {
+		return true, nil
```

Comment:
The early return here is surprising. I get what it's doing, but as a reader, I'd expect to something more like

```
_, err := dropletByID(context.TODO(), i.client, id)
if err != nil {
    godoErr, ok := err.(*godo.ErrorResponse)
    if !ok {
        return false, fmt.Errorf("unexpected error type from godo: %T; %v", err, err)
    }

    if godoErr.Response. StatusCode != http.StatusNotFound {
        return false, fmt.Errorf("error checking if instance exists: %v", err)
    }
    return false, nil
}

return true, nil
```

Reply:
hmm, interesting. I always tend try to reduce the amount of code thats indented for readability so it made more sense here to return early. Though it can be misleading cause we're so used to seeing `err != nil` 😛

Reply:
I like reducing indentation :-). And you're right, it is because we're used to seeing `err != nil`. Clearly, this is correct, and I don't feel strongly about changing it, but I did stumble over it when I first read it.. take that for what it's worth.